src/material/datepicker/date-range-input.scss

```diff
@@ -0,0 +1,91 @@
+@import '../core/style/vendor-prefixes';
+
+$mat-date-range-input-separator-spacing: 4px;
+$mat-date-range-input-part-max-width: calc(50% - #{$mat-date-range-input-separator-spacing});
+
+// Host of the date range input.
+.mat-date-range-input {
+  display: block;
+  width: 100%;
+}
+
+// Inner container that wraps around all the content.
+.mat-date-range-input-container {
+  display: flex;
+  align-items: center;
+}
+
+// Text shown between the two inputs.
+.mat-date-range-input-separator {
+  margin: 0 $mat-date-range-input-separator-spacing;
+}
+
+// Underlying input inside the range input.
+.mat-date-range-input-inner {
+  // Reset the input so it's just a transparent rectangle.
```

Comment:
It might be worth putting a common class for this in mat-core, since I assume `MatInput` has to jump through all the same hoops

Reply:
I agree that we should have a common place for it, but it might be a lot of styles that would be in the bundle, even if the form field isn't used.

Also this is a slightly stripped down version of the one on `.mat-input` because it doesn't have to deal with the different input types.

Reply:
I wonder if we could have something like MDC's emit-once. That would let us `@include` mixins where needed but only actually emit the rule one time. That could probably be a separate bigger refactoring though

Reply:
I'll leave it as is for now. I imagine that I'll have to do a cleanup pass over it before merging into master anyway.